python/ray/air/tests/execution/test_tracked_actor.py

```diff
@@ -285,6 +287,79 @@ def __init__(self):
     assert actor_manager.num_live_actors == 0
 
 
+@pytest.mark.parametrize(
+    "resource_manager_cls", [FixedResourceManager, PlacementGroupResourceManager]
+)
+@pytest.mark.parametrize("start_thread", [False, True])
+def test_stop_actor_custom_future(
+    ray_start_4_cpus, tmp_path, cleanup, resource_manager_cls, start_thread
+):
+    """If we pass a custom stop future, the actor should still be shutdown by GC.
+
+    This should also be the case when we start a thread in the background, as we
+    do e.g. in Ray Tune's function runner.
+    """
```

Comment:
Is this test for a different bug that was fixed?

Reply:
This worked well already, but it was a missing test case in the actor manager API (using a custom stop future). I wanted to ensure that there was no leak because we're using a custom stop future rather than `__ray_terminate__`.